src/type.coffee

```diff
@@ -49,13 +49,20 @@ module.exports = class Type extends Emitter
       resource
 
   _getByIDs: (ids, otherArgs...) ->
-    requests = for id in ids
+    {toQuery, resources} = ids.reduce(((accum, id) =>
       if id of @_resourcesCache and (otherArgs.length is 0 or not otherArgs[0]?)
-        Promise.resolve @_resourcesCache[id]
+        accum.resources.push Promise.resolve @_resourcesCache[id]
       else
-        @_client.get(@_getURL(id), otherArgs...).then ([resource]) ->
-          resource
-    Promise.all requests
+        accum.toQuery.push id
+      accum),
+      {toQuery: [], resources: []})
+    if toQuery.length is 1
+      resources.push @_client.get(@_getURL(toQuery[0]), otherArgs...).then ([resource]) ->
```

Comment:
Can't the consumer just do the ordering if they need it? It saves us an extra iteration since I'd image order won't be important in most cases.